lib/src/components/Toast/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { toast as toastRHT } from 'react-hot-toast'
+import type { ToastPosition } from 'react-hot-toast'
+
+import type { TextProps } from '@/components/Text'
+import { classNames } from '@/utils'
+
+import { Growl } from './components/Growl'
+import { Snackbar, SnackbarAction } from './components/Snackbar'
+import toastStyles from './toast.module.scss'
+import type { ToastOptions } from './types'
+
+const cx = classNames(toastStyles)
+
+const Title: React.FC<TextProps> = ({ children, ...rest }) => (
+  <p className={cx('title', rest.className)} {...rest}>
+    {children}
+  </p>
+)
+
+export const toast = (component: JSX.Element, options?: ToastOptions) => {
+  const name =
+    typeof component === 'string'
+      ? undefined
+      : component?.type?.displayName || component?.type?.name
```

Comment:
this should work: `const name = 'type' in component ? component.type.displayName || component.type.name : undefined` and is more readable in my opinion